tests/test_cli_compile.py

```diff
@@ -2735,9 +2737,383 @@ def test_all_extras_fail_with_extra(fake_dists, runner, make_module, fname, cont
+        ),
+        # Test also that the via section is correct
+        # Since this is rather lengthy and we are concerned mostly with the case where more than
+        # one build-system hook requests the same dependency we only test it with this last
+        # parameter.
+        (
+            ["editable", "sdist", "wheel"],
+            ["--all-extras"],
+            {
+                "setuptools": Dependency(
+                    "*",
+                    [
+                        "small-fake-with-build-deps (pyproject.toml::build-system.requires)",
+                    ],
+                ),
+                "fake-static-build-dep": Dependency(
+                    "0.1",
+                    [
+                        "small-fake-with-build-deps (pyproject.toml::build-system.requires)",
+                    ],
+                ),
+                "fake-dynamic-build-dep-for-all": Dependency(
+                    "0.2",
+                    [
+                        (
+                            "small-fake-with-build-deps"
+                            " (pyproject.toml::build-system.backend::editable)"
+                        ),
+                        "small-fake-with-build-deps (pyproject.toml::build-system.backend::sdist)",
+                        "small-fake-with-build-deps (pyproject.toml::build-system.backend::wheel)",
+                    ],
+                ),
+                "fake-dynamic-build-dep-for-sdist": Dependency(
+                    "0.3",
+                    [
+                        "small-fake-with-build-deps (pyproject.toml::build-system.backend::sdist)",
+                    ],
+                ),
+                "fake-dynamic-build-dep-for-wheel": Dependency(
+                    "0.4",
+                    [
+                        "small-fake-with-build-deps (pyproject.toml::build-system.backend::wheel)",
+                    ],
+                ),
+                "fake-dynamic-build-dep-for-editable": Dependency(
+                    "0.5",
+                    [
+                        (
+                            "small-fake-with-build-deps"
+                            " (pyproject.toml::build-system.backend::editable)"
+                        ),
+                    ],
+                ),
+                "small-fake-a": Dependency(
+                    "0.1",
+                    [
+                        "small-fake-with-build-deps (setup.py)",
+                    ],
+                ),
+                "small-fake-b": Dependency(
+                    "0.2",
+                    [
+                        "small-fake-with-build-deps (setup.py)",
+                    ],
+                ),
+                "wheel": Dependency(
+                    "*",
+                    [
+                        "small-fake-with-build-deps (pyproject.toml::build-system.backend::wheel)",
+                    ],
+                ),
+            },
+        ),
+    ),
+)
+def test_build_distribution(
```

Comment:
This test is the most time-consuming and has parameterizations, which, while great, are too expensive for integration tests. What if we unit-test them instead? This could save considerable time by eliminating the need to build sdists and wheels. What do you think?


<details>

<summary>Test durations</summary>

```
46.83s call     tests/test_cli_compile.py::test_build_distribution[backtracking resolver-distributions2-other_options2-expected_deps2]
44.45s call     tests/test_cli_compile.py::test_build_distribution[legacy resolver-distributions0-other_options0-expected_deps0]
31.00s call     tests/test_cli_compile.py::test_build_distribution[backtracking resolver-distributions3-other_options3-expected_deps3]
27.38s call     tests/test_cli_compile.py::test_build_distribution[legacy resolver-distributions1-other_options1-expected_deps1]
26.07s call     tests/test_cli_compile.py::test_build_distribution[backtracking resolver-distributions1-other_options1-expected_deps1]
...
```

</details>

Reply:
I have rewritten the tests to prioritize speed. Let me know what you think.

Reply:
Did we loose clear test cases with this rewrite?